### PR TITLE
Ability to add locales for quest reward text

### PR DESCRIPTION
**Pull request by AlcDenat**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**
Adds a new table quest_offer_reward_locale for quest ending text

**Target branch(es):** 3.3.5/master
- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)
Does build and works ingame.

**Known issues and TODO list:** (add/remove lines as needed)
none

### CODE REVIEW

File: src/server/game/Entities/Creature/GossipDef.cpp [hunk 1]
```diff
@@ -624,13 +624,17 @@ void PlayerMenu::SendQuestGiverOfferReward(Quest const* quest, ObjectGuid npcGUI
...
-            ObjectMgr::GetLocaleString(questTemplateLocale->OfferRewardText,    locale, questOfferRewardText);
-            ObjectMgr::GetLocaleString(questTemplateLocale->PortraitGiverText,  locale, portraitGiverText);
-            ObjectMgr::GetLocaleString(questTemplateLocale->PortraitGiverName,  locale, portraitGiverName);
+            ObjectMgr::GetLocaleString(questTemplateLocale->LogTitle, locale, questTitle);
+            ObjectMgr::GetLocaleString(questTemplateLocale->PortraitGiverText, locale, portraitGiverText);
+            ObjectMgr::GetLocaleString(questTemplateLocale->PortraitGiverName, locale, portraitGiverName);
             ObjectMgr::GetLocaleString(questTemplateLocale->PortraitTurnInText, locale, portraitTurnInText);
             ObjectMgr::GetLocaleString(questTemplateLocale->PortraitTurnInName, locale, portraitTurnInName);
         }
+
```
Comment (Exodius) [inline src/server/game/Entities/Creature/GossipDef.cpp]: Whitespace.

File: src/server/game/Quests/QuestDef.h [hunk 2]
```diff
@@ -264,6 +263,11 @@ struct QuestObjectivesLocale
     StringVector Description;
 };
 
+struct QuestOfferRewardLocale
+{
+	StringVector RewardText;
```
Comment (Exodius) [inline src/server/game/Quests/QuestDef.h]: tabs

File: sql/updates/world/master/9999_99_99_99_world.sql [hunk 3]
```diff
@@ -0,0 +1,8 @@
+DROP TABLE IF EXISTS `quest_offer_reward_locale`;
+CREATE TABLE `quest_offer_reward_locale` (
+  `ID` int(10) unsigned NOT NULL DEFAULT '0',
+  `locale` varchar(4) NOT NULL,
+  `RewardText` text,
+  `VerifiedBuild` smallint(6) NOT NULL DEFAULT '0',
+  PRIMARY KEY (`ID`,`locale`)
+)
```
Comment (joschiwald) [inline sql/updates/world/master/9999_99_99_99_world.sql]: you have to migrate data in locales_quest

Comment (AlcDenat) [inline sql/updates/world/master/9999_99_99_99_world.sql]: There's no such table on the master branch, only on 3.3.5.
On 3.3.5 it looks grouped together in one table while in master if there is a table with enUS Text, there is (mostly) a second table *_locales with text locales.

Comment (joschiwald) [inline sql/updates/world/master/9999_99_99_99_world.sql]: wtf, who dropped all of our locales ...

**Comment by Aokromes** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Spells/SpellEffects.cpp:5222:43: fatal error: 
      unused parameter 'effIndex' [-Wunused-parameter]
void Spell::EffectTitanGrip(SpellEffIndex effIndex)
                                          ^
1 error generated.
make[2]: *** [src/server/game/CMakeFiles/game.dir/Spells/SpellEffects.cpp.o] Error 1

**Comment by Eredar** [comment]
What about greetings_locale?

**Comment by tkrokli** [comment]
@Eredar : feel free to start a PR for it, or open an issue if it is missing (if there are no open issues about it already).

**Comment by AlcDenat** [comment]
quest_greeting and quest_request_items are both missing their *_locale table, I'll add them later.
Problem is, I know no DB or site that has them locales for those text - Noone is tracking them.

**Comment by joschiwald** [comment]
we had translations in our old 6.x TDB, i can migrate these datas for you

**Comment by AlcDenat** [comment]
Any way to find these old translations or can you push them? Also what tables are needed, I'd like to get as much localized as possible.

**Comment by joschiwald** [comment]
I'll do it for you

**Comment by joschiwald** [comment]
Ok, we have no data in our TDB 6.01 (last db release where locales_quest exists), in 3.3.5 db we have very few entries, but does not deserved any migration

my bad, i thought we had bruteforced all data, but were not bruteforcable

**Pull request closed without merging**